spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/ConfigMapPropertySource.java

```diff
@@ -89,12 +95,43 @@ private static String getName(KubernetesClient client, String name,
 	}
 
 	private static Map<String, Object> getData(KubernetesClient client, String name,
-			String namespace, Environment environment) {
+			String namespace, Environment environment, boolean enableVersioning) {
 		try {
 			Map<String, Object> result = new LinkedHashMap<>();
-			ConfigMap map = StringUtils.isEmpty(namespace)
-					? client.configMaps().withName(name).get()
-					: client.configMaps().inNamespace(namespace).withName(name).get();
+			ConfigMap map = null;
+			if (enableVersioning) {
+				String version = environment.getProperty("info.app.version");
```

Comment:
Where is this coming from?

Reply:
From build-info.properties. So it would require to add goal build-info to spring-boot-maven-plugin:

```
<plugin>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-maven-plugin</artifactId>
	<executions>
		<execution>
			<goals>
				<goal>build-info</goal>
			</goals>
		</execution>
	</executions>
</plugin>
```